src/core/openthread.cpp

```diff
@@ -181,6 +181,11 @@ void otGetFactoryAssignedIeeeEui64(otInstance *aInstance, otExtAddress *aEui64)
     otPlatRadioGetIeeeEui64(aInstance, aEui64->m8);
 }
 
+void otGetHashMacAddress(otInstance *, otExtAddress *aHashMacAddress)
+{
+    sThreadNetif->GetMac().GetHashMacAddress(static_cast<Mac::ExtAddress *>(aHashMacAddress));
```

Comment:
ah, that's fine, it's an implicit conversion between base class and inherited class. You are correct. Please ignore this.